src/Core/Components/InputFile/FluentInputFile.razor.cs

```diff
@@ -201,13 +201,13 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
         {
             Module ??= await JSRuntime.InvokeAsync<IJSObjectReference>("import", JAVASCRIPT_FILE.FormatCollocatedUrl(LibraryConfiguration));
 
-            if (!string.IsNullOrEmpty(AnchorId))
-            {
-                await Module.InvokeVoidAsync("attachClickHandler", AnchorId, Id);
-            }
-
             _containerInstance = await Module.InvokeAsync<IJSObjectReference>("initializeFileDropZone", _containerElement, _inputFile!.Element);
         }
+
+        if (!string.IsNullOrEmpty(AnchorId) && Module is not null)
+        {
+            await Module.InvokeVoidAsync("attachClickHandler", AnchorId, Id);
+        }
```

Comment:
Did we not discuss that this would stay inside the 'firstRender-block'?

Reply:
@vnbaaij it can'T that's the whole point of this. In order for the event to be reattached it needs to be done outside the if(firstRender) block

Reply:
Right!